### PR TITLE
feat: Add Strimzi KafkaConnect CRD custom health checks

### PR DESCRIPTION
**Pull request by chris-vest**
This is a PR to contribute a custom health check to ArgoCD for Strimzi's KafkaConnect CRD. My team had an issue where the `KafkaConnect` resource would be forever stuck in a `Progressing` state, so we created this health check.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I've signed the CLA and my build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

### CODE REVIEW

File: resource_customizations/kafka.strimzi.io/KafkaConnect/health.lua [hunk 1]
```diff
@@ -0,0 +1,21 @@
+hs = {}
+if obj.status ~= nil then
+  if obj.status.conditions ~= nil then
+    for i, condition in ipairs(obj.status.conditions) do
+      if condition.type == "Ready" and condition.status == "False" then
```
Comment (jannfis) [inline resource_customizations/kafka.strimzi.io/KafkaConnect/health.lua:5]: Hm, I am wondering whether the state should be really set to `Degraded` here. Could it be that that the real state is still progressing, when `Ready` condition is `False`? Is there another condition to check whether the operator has failed where we can set `Degraded` status?

**Review by jannfis — CHANGES_REQUESTED** [review]
Hey @chris-vest, thanks a lot for your PR, much appreciated!

All looks fine so far, I have just minor comment (or rather a question), please see below and let me know your opinion on that. Could be that I'm wrong on that, I don't know the Strimzi operator in detail.

File: resource_customizations/kafka.strimzi.io/KafkaConnect/health.lua (hunk 1 above)
Comment (chris-vest) [reply resource_customizations/kafka.strimzi.io/KafkaConnect/health.lua:5]: I'll have a dig into what the CRD exposes! Thanks for the comment.

Comment (chris-vest) [reply resource_customizations/kafka.strimzi.io/KafkaConnect/health.lua:5]: I'm looking into this, having a chat with the Strimzi guys.

Comment (chris-vest) [reply resource_customizations/kafka.strimzi.io/KafkaConnect/health.lua]: @jannfis Updated!

Comment (jannfis) [reply resource_customizations/kafka.strimzi.io/KafkaConnect/health.lua]: Great, thank you @chris-vest! One last request: Can you please rebase your PR with most recent master, and then I'm going to merge it if all tests pass.

Comment (chris-vest) [reply resource_customizations/kafka.strimzi.io/KafkaConnect/health.lua]: Sure!

Comment (chris-vest) [reply resource_customizations/kafka.strimzi.io/KafkaConnect/health.lua]: Should be rebased!

**Review by jannfis — APPROVED** [review]
LGTM, thank you!

**Pull request merged**